### DISCUSSION TITLE
Any luck using gsSQL as a custom function?

### DISCUSSION THREAD

**Discussion by demmings** [demmings/gsSQL#2] [Q&A]
Any real world feedback would be appreciated.
I look forward to any questions that you may have.

Thanks Chris.

**Comment by opinsky** [comment]
many, many thanks for that. heavy user of gsheets query, arrays & filter here. will likely incorporate in many of my projects. curious about the performance, and happy to contribute (to documentation, mostly)

**Comment by demmings** [comment]
Hi opinsky.

I am most interested in any areas where you encounter issues getting up and going.  Things that are obvious to me and missed in the doc, are definitely things I would like to improve upon.  As for speed, definitely the first use of the day is slower, after that if you can set longer cache times for tables/ranges that don't change often, it is much master the second time round.  In the end, the single biggest component of the total time to run is calling **getValues()**.  So when Google is sluggish, **gsSQL()** is sluggish.

My biggest headaches after writing, was trying to convert long and convoluted QUERY statements written long enough ago that I could not remember exactly what I was trying to do into an SQL statement.

Thanks for you interest.

Chris.

**Comment by opinsky** [comment]
Thanks Chris, for the prompt reply.

From a documentation perspective, I think a demo sheet in google sheets
would be a great addition. As well as an easier way to install it. I can
help on both

Oren

**Reply by demmings** [reply]
Oren,

Here is the sheet I used to help make by README.md.  This could be a good starting point (not sure how to make it so you can't edit original, but yet be able to make your own copy for experimenting).

https://docs.google.com/spreadsheets/d/1Zmyk7a7u0xvICrxen-c0CdpssrLTkHwYx6XL00Tb1ws/edit?usp=sharing
